engines/terraform/resolve.go

```diff
@@ -84,13 +57,118 @@ func (td *TerraformDeployment) resolveValue(intentName string, value interface{}
 	}
 }
 
+func (td *TerraformDeployment) resolveToken(intentName string, specRef *SpecReference) (interface{}, error) {
+	switch specRef.Source {
+	case "infra":
+		if len(specRef.Path) < 2 {
+			return nil, fmt.Errorf("infra token requires at least 2 path components")
+		}
+
+		refName := specRef.Path[0]
+		attribute := strings.Join(specRef.Path[1:], ".")
+
+		infraResource, err := td.resolveInfraResource(refName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve infrastructure resource %s: %w", refName, err)
+		}
+
+		result := infraResource.Get(jsii.String(attribute))
+		if result == nil {
+			return nil, fmt.Errorf("attribute '%s' not found on infrastructure resource '%s' (type: %s)", attribute, refName, *infraResource.Node().Id())
+		}
+		return result, nil
+
+	case "self":
+		if len(specRef.Path) == 0 {
+			return nil, fmt.Errorf("references 'self' without specifying a variable. All references must include a variable name e.g. ${self.my_var}")
+		}
+
+		varName := specRef.Path[0]
+
+		tfVariable, ok := td.instancedTerraformVariables[intentName][varName]
+		if !ok {
+			return nil, fmt.Errorf("variable %s does not exist for provided blueprint", varName)
+		}
+
+		if len(specRef.Path) > 1 {
+			attribute := strings.Join(specRef.Path[1:], ".")
+			return cdktf.Fn_Lookup(tfVariable.Value(), jsii.String(attribute), nil), nil
+		}
+		return tfVariable.Value(), nil
+
+	case "var":
+		if len(specRef.Path) < 1 {
+			return nil, fmt.Errorf("var token requires at least 1 path component")
```

Comment:
Can we determine the exact original token/reference used in this case? It would be nice to say something like:

"'${var.}' doesn't contain a valid variable reference"

or `${var}` if they didn't include the dot.

Reply:
When we create the spec reference we should store the original/raw value for this. So its referencable as `specRef.Raw` or something similiar